persistent-template/Database/Persist/TH.hs

```diff
@@ -490,9 +494,10 @@ sqlSettings = mkPersistSettings $ ConT ''SqlBackend
 
 recNameNoUnderscore :: MkPersistSettings -> HaskellName -> HaskellName -> Text
 recNameNoUnderscore mps dt f
-  | mpsPrefixFields mps = lowerFirst (unHaskellName dt) ++ upperFirst ft
-  | otherwise           = lowerFirst ft
+  | mpsPrefixFields mps = modifier (lowerFirst $ unHaskellName dt) (upperFirst ft)
```

Comment:
I think that we could have a more consistent and reliable ordering if we did:

```
lowerFirst (modifier (unHaskellName dt) (upperFirst ft))
```

This also protects the user against providing an invalid field name. Consider if their modifier function was:

```haskell
\entity field -> "SchemaThree" <> entity <> field
```

For a `User; name String` model, this would make a field name of `SchemaThreeuserName`. Which isn't a valid Haskell identifier for a record field. But with that re-ordering, we'd get `schemaThreeUserName`, which is valid.

Reply:
Good idea, will fix